### PR TITLE
stop() balancedconsumer in response to errors in start()

### PR DESCRIPTION
**Pull request by emmett9001**
This pull request safely stops the `BalancedConsumer` if errors are encountered during `start()`.

Connected to #328

### CODE REVIEW

**Comment by emmett9001** [comment]
@yungchin I'm not sure this is completely correct, but it's at least a start. Let me know if there are other things you think need to change here.

File: pykafka/balancedconsumer.py [hunk 1]
```diff
@@ -259,16 +260,19 @@ def stop(self):
...
+                # of our partitions until consumption has really been halted
+                self._zookeeper.stop()
+            else:
+                self._remove_partitions(self._get_held_partitions())
+                self._zookeeper.delete(self._path_self)
+                # additionally we'd want to remove watches here, but there are no
+                # facilities for that in ChildrenWatch - as a workaround we check
+                # self._running in the watcher callbacks (see further down)
+        except NoNodeError:
+            log.warning("Error encountered while closing ZooKeeper connection")
```
Comment (yungchin) [inline pykafka/balancedconsumer.py]: Minor nitpick: this makes it sound as if something unexpected happened, when in fact we know why this happens. If we'd wrap the `try .. except` around the `delete(_path_self)` line only, I'd argue the error would be so specific that you could just `pass` without logging it?

Comment (yungchin) [inline pykafka/balancedconsumer.py]: (Just to check, I created a `BalancedConsumer` with an external `KazooClient` instance, and then called `stop()` on it twice - indeed the other lines here don't raise exceptions, just that one.)

**Comment by yungchin** [comment]
Apart from the one inline comment, I've nothing to complain. I think this is all correct, and much better than preventing `stop()` if not `_running` (which this did before this patch).

**Pull request merged**